pkg/core/native/designate.go

```diff
@@ -184,9 +184,17 @@ func (s *Designate) OnPersist(ic *interop.Context) error {
 
 // PostPersist implements the Contract interface.
 func (s *Designate) PostPersist(ic *interop.Context) error {
-	cache := ic.DAO.GetRWCache(s.ID).(*DesignationCache)
-	if !cache.rolesChangedFlag {
-		return nil
+	s.NotifyServices(ic.DAO, false)
+	return nil
+}
+
+// NotifyServices notifies dependent services about node roles. It does not
+// check whether roles were updated in the last block. It expects the cache to
+// be initialized and RW.
+func (s *Designate) NotifyServices(dao *dao.Simple, force bool) {
```

Comment:
You can hide the `force` from public method if all callers use it this way (and postpersist can handle deal with caches internally).